pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -1661,22 +1666,35 @@ private void makeStreamConsumer(String reason) {
       closePartitionGroupConsumer();
     }
     _segmentLogger.info("Creating new stream consumer for topic partition {} , reason: {}", _clientId, reason);
-    _partitionGroupConsumer =
-        _streamConsumerFactory.createPartitionGroupConsumer(_clientId, _partitionGroupConsumptionStatus);
-    _partitionGroupConsumer.start(_currentOffset);
+    try {
+      _partitionGroupConsumer =
+          _streamConsumerFactory.createPartitionGroupConsumer(_clientId, _partitionGroupConsumptionStatus);
+      _partitionGroupConsumer.start(_currentOffset);
+    } catch (Exception e) {
+      _segmentLogger.error("Faced exception while trying to recreate stream consumer for topic partition {} reason {}",
+          _clientId, reason, e);
+      _serverMetrics.addMeteredTableValue(_clientId, ServerMeter.STREAM_CONSUMER_CREATE_EXCEPTIONS, 1L);
+      throw e;
+    }
   }
 
   /**
    * Checkpoints existing consumer before creating a new consumer instance
    * Assumes there is a valid instance of {@link PartitionGroupConsumer}
    */
   private void recreateStreamConsumer(String reason) {
-    _segmentLogger.info("Recreating stream consumer for topic partition {}, reason: {}", _clientId, reason);
-    _currentOffset = _partitionGroupConsumer.checkpoint(_currentOffset);
-    closePartitionGroupConsumer();
-    _partitionGroupConsumer =
-        _streamConsumerFactory.createPartitionGroupConsumer(_clientId, _partitionGroupConsumptionStatus);
-    _partitionGroupConsumer.start(_currentOffset);
+    try {
+      _segmentLogger.info("Recreating stream consumer for topic partition {}, reason: {}", _clientId, reason);
+      _currentOffset = _partitionGroupConsumer.checkpoint(_currentOffset);
+      closePartitionGroupConsumer();
```

Comment:
If they throw an exception, the consumer wouldn't get created right? And `_currentOffset = _partitionGroupConsumer.checkpoint(_currentOffset);` does throw a runtime exception (at least in code).

Reply:
if they throw an exception, is the old consumer closed? if not, the consumer is still alive, right?

Reply:
I guess to be sure we should re-throw the exception in the catch part.

Reply:
can you please help me understand why re-throw the exception can address the potential issue I mentioned above?

Reply:
In
```
     1.  _currentOffset = _partitionGroupConsumer.checkpoint(_currentOffset);
     2.  closePartitionGroupConsumer();
     3.  _partitionGroupConsumer =
          _streamConsumerFactory.createPartitionGroupConsumer(_clientId, _partitionGroupConsumptionStatus);
      _partitionGroupConsumer.start(_currentOffset);
```
If 1st or 2nd throws an exception, 3rd won't be executed - This is a valid case of incrementing this metric, no?

>then the exception is not related to consumer creation.

This metric tracks if a consumer cannot be created for any reason. In this case as well, the consumer isn't created.

Please correct me if I am wrong.

Reply:
If the 1st or 2nd line throws an exception,
1. that means the old consumer is still running, may be it can still be used for consuming data?
2. the error is not really about creating consumer, in this case, maybe we don't want to increase the metric, WDYT?